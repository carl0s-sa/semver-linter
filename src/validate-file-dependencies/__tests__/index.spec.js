const validateFileDependencies = require('../')

const releaseCandidateMocks = {
	"package-that-throws": "4.2.0-rc1",
	"and-package-that-throws": "4.2.0-rc.1",
	"other-package-that-throws": "4.0.4-rc.2",
	"another-package-that-throws": "^4.2.9-rc.1",
	"yaptt": "0.10.1-rc.c4f3f999.8+c4f3f999",
}

describe('Given a valid JSON object', () => {
	let validJSON
	beforeEach(() => {
		validJSON = {
			"dependencies": {
				"package": "1.0.0",
			},
			"devDependencies": {
				"package": "1.0.0",
			},
		}
		validJSON = JSON.stringify(validJSON)
	})

	describe('with no non-stable versions', () => {
		it('should return 0', () => {
			expect(validateFileDependencies(validJSON)).toBe(0)
		})
	})

	describe('with non-stable versions', () => {
		beforeEach(() => {
			validJSON = {
				dependencies: {
					...validJSON.dependencies,
					...releaseCandidateMocks,
				},
				devDependencies: {
					...validJSON.devDependencies,
					...releaseCandidateMocks,					
				},
			}
			validJSON = JSON.stringify(validJSON)
		})

		it('should throw an error naming every package with non-stable version', () => {
			expect(() => validateFileDependencies(validJSON)).toThrowErrorMatchingSnapshot()
		})
	})

})

describe('Given an invalid JSON object', () => {
	let invalidJSON = "asd"

	it('should throw an error', () => {
		expect(() => validateFileDependencies(invalidJSON)).toThrowErrorMatchingSnapshot()
	})
})