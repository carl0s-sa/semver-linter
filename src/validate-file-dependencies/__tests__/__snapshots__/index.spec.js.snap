// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a valid JSON object with non-stable versions should throw an error naming every package with non-stable version 1`] = `
"Non stable version found for package package-that-throws: 4.2.0-rc1
Non stable version found for package and-package-that-throws: 4.2.0-rc.1
Non stable version found for package other-package-that-throws: 4.0.4-rc.2
Non stable version found for package another-package-that-throws: ^4.2.9-rc.1
Non stable version found for package yaptt: 0.10.1-rc.c4f3f999.8+c4f3f999
Non stable version found for package package-that-throws: 4.2.0-rc1
Non stable version found for package and-package-that-throws: 4.2.0-rc.1
Non stable version found for package other-package-that-throws: 4.0.4-rc.2
Non stable version found for package another-package-that-throws: ^4.2.9-rc.1
Non stable version found for package yaptt: 0.10.1-rc.c4f3f999.8+c4f3f999"
`;

exports[`Given an invalid JSON object should throw an error 1`] = `"Invalid package.json format"`;
